core/domain/calculation_registry_test.py

```diff
@@ -35,3 +38,14 @@ def test_get_calculation_by_id(self) -> None:
         with self.assertRaisesRegex(
             TypeError, '\'a\' is not a valid calculation id.'):
             calculation_registry.Registry.get_calculation_by_id('a')
+
+    def test_get_calculation_by_id_when_calculations_dict_is_not_empty(
+        self) -> None:
+        # AnswerFrequencies is present in calculations_dict,
+        # because AnswerFrequencies is inserted in calculations_dict in
+        # test_get_calculation_by_id_when_calculations_dict_is_empty test case.
```

Comment:
Hm, ideally we want our unit tests built such that they do not depend on each other in anyway. Here it seems like there is some link between two test cases. It seems like there is some sort of caching going on in `get_calculation_by_id`, and we want to cover a specific test. Can we keep the original test as-is and have a new test where we call get_calculation_id with a non-existent "calculation_id" and then call it again with the same calculation_id and expect it to exist. Would that cover the necessary statements?

Reply:
Hi @kevintab95 , instead of writing separate test case its better to add the statement in the existing statement is the best way we can do.
I have already showed you both types of ways.
in one test case and
in separate test case
And  one more thing previous unit test case will always have a impact on this because it is filling in dictionary.

Reply:
Hi @kashishbhandula, my suggestion still stands. We expect test cases to be independent and that they test a specific behaviour -- it shouldn't cover lines as a side-effect. We have discussed this in the wiki pages for testing in general. You might find these useful:

https://github.com/oppia/oppia/wiki/Backend-tests#guidelines-for-writing-good-tests

https://github.com/oppia/oppia/wiki/End-to-End-Tests#independence

https://github.com/oppia/oppia/wiki/Frontend-tests#tests-should-work-in-any-order

(Note that even though these wiki pages may talk about e2e / frontend tests, the concepts and practices discussed are universal and are applicable for all kinds of unit tests)